src/shutdown/umount.c

```diff
@@ -617,35 +624,49 @@ static int remount_with_timeout(MountPoint *m, bool last_try) {
                                        "Failed to remount '%s' read-only: %m",
                                        m->path);
 
+                (void) write(pfd[1], &r, sizeof(r)); /* try to send errno up */
                 _exit(r < 0 ? EXIT_FAILURE : EXIT_SUCCESS);
         }
 
+        pfd[1] = safe_close(pfd[1]);
+
         r = wait_for_terminate_with_timeout(pid, DEFAULT_TIMEOUT_USEC);
-        if (r == -ETIMEDOUT) {
+        if (r == -ETIMEDOUT)
                 log_error_errno(r, "Remounting '%s' timed out, issuing SIGKILL to PID " PID_FMT ".", m->path, pid);
-                (void) kill(pid, SIGKILL);
-        } else if (r == -EPROTO)
-                log_debug_errno(r, "Remounting '%s' failed abnormally, child process " PID_FMT " aborted or exited non-zero.", m->path, pid);
-        else if (r < 0)
+        else if (r == -EPROTO) {
+                /* Try to read error code from child */
+                if (read(pfd[0], &r, sizeof(r)) == sizeof(r))
+                        log_debug_errno(r, "Remounting '%s' failed abnormally, child process " PID_FMT " failed: %m", m->path, pid);
+                else
+                        log_debug_errno(r, "Remounting '%s' failed abnormally, child process " PID_FMT " aborted or exited non-zero.", m->path, pid);
```

Comment:
In this case, `r` may be invalid, especially 0 and `log_debug_errno()` may trigger assertion.
I think it is better to read error code with another variable. 
Or use `EPROTO` explicitly.

Reply:
#25997.